src/hotspot/share/oops/instanceKlass.cpp

```diff
@@ -2402,9 +2402,9 @@ void InstanceKlass::metaspace_pointers_do(MetaspaceClosure* it) {
 
 void InstanceKlass::remove_unshareable_info() {
 
-  if (can_be_verified_at_dumptime()) {
-    // Set the old class bit.
-    set_is_shared_old_klass();
+  if (!can_be_verified_at_dumptime()) {
+    // Clear the _verified_at_dump_time bit.
+    clear_verified_at_dump_time();
```

Comment:
I think it's better to change this to 

```
if (can_be_verified_at_dumptime()) {
  // Remember this so we can avoid walking the hierarchy at runtime.
  set_verified_at_dump_time();
```

and do not set this bit inside `Klass::Klass()`. This way, all the code that changes this bit are kept in a single place.

Reply:
Thanks Ioi.

I've pushed an update with your suggestion.